temporal-sdk/src/main/java/io/temporal/worker/WorkerFactory.java

```diff
@@ -279,14 +277,28 @@ public WorkflowClient getWorkflowClient() {
   public synchronized void shutdown() {
     log.info("shutdown");
     state = State.Shutdown;
-    if (stickyPoller != null) {
-      stickyPoller.shutdown();
-      // To ensure that it doesn't get new tasks before workers are shutdown.
-      stickyPoller.awaitTermination(1, TimeUnit.SECONDS);
-    }
-    for (Worker worker : workers.values()) {
-      worker.shutdown();
-    }
+    ShutdownManager shutdownManager = new ShutdownManager();
```

Comment:
> Should this not wait for completion of shutdown?
It will not.
The previous implementation also didn't.

> Maybe you want to follow Java's lead here and have this shutdown, a shutdownNow (though ours doesn't have to return runnables), and an awaitTermination.

Exactly! We already do.
WorkerFactory already has non-blocking shutdown() and awaitTermination, like Java's Executor. This PR doesn't change it.

Reply:
So long as, from the outside, those three calls work including `awaitTermination`, no prob.